### PR TITLE
Hotel Room Booking

### PR DESCRIPTION
**Pull request by gaks1**
**Description:**
This PR implements the backend for a hotel room reservation system. The backend provides APIs to handle user authentication, manage hotel room listings, enable reservations, and display user reservations. The frontend React app (referenced in the frontend PR) can interact with these APIs to offer a complete hotel room reservation experience.

**Core Features:**

1. **User Authentication:**
    - Implemented a proper authenticated login for teams with 5 members.
    - Users can log in to the system using their username.
2. **Hotel Room Management:**
    - Created APIs to manage hotel room listings.
    - Added functionalities to add, update, and delete hotel room information.
3. **Reservation System:**
    - Developed APIs to handle room reservations.
    - Users can reserve hotel rooms by selecting dates and other necessary details.
    - Reservations are stored in the database for future reference.
4. **User Reservations:**
    - Created APIs to fetch a user's reservations.
    - Users can view a list of their reservations, including details like room name, dates, and other relevant information.
5. **API Documentation:**
    - Provided comprehensive API documentation for easy understanding and integration with the frontend app.

**Additional Features (For Teams with 3-4 & 5 Members):**

1. **Delete Hotel Room:**
    - Added API to delete hotel room listings.
    - When called, the selected hotel room will be removed from the available room listings.
2. **Reservation Autofill:**
    - Implemented APIs to autofill user details (username and selected item) when making a reservation
    - Added a feature to select a duration of stay while making a reservation.
    - Users can choose from available cities for their hotel stay
3. **Testing:**
    - Conducted extensive testing of all APIs and functionalities to ensure reliability and proper error handling.

[React](https://github.com/Jolak5/hotel-booking-app-frontend/pull/36) Front End

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,182 @@
-# hotel-booking-app-backend
\ No newline at end of file
+<a name="readme-top"></a>
```
Comment (Meltrust) [inline README.md:1]: - [ ] When I run the tests with the `rspec ./spec` command, all of them fail:

image

If this is a bug, kindly correct it.  If this is not a bug, please include specific instructions in your readme about how to setup and run your tests correctly. 👍

Comment (Meltrust) [inline README.md:1]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

File: README.md [hunk 2]
```diff
@@ -1 +1,182 @@
...
+
+> cd hotel-booking-app-backend
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
+
+Runs the app in the development mode.
```
Comment (Meltrust) [inline README.md]: - [ ] These instructions pertain the frontend project, not the backend.  Kindly correct these instructions.

File: README.md [hunk 3]
```diff
@@ -1 +1,182 @@
...
+
+In order to run this project you only need:
+
+- latest web-browser
+
+### Install
+
+To install this project in your local machine use
+
+> gem install
```
Comment (Meltrust) [inline README.md]: - [ ] This command won't work.  The correct command is `bundle install`.  Kindly correct it.

File: README.md [hunk 4]
```diff
@@ -1 +1,182 @@
...
+
+### `npm start`
+
+Runs the app in the development mode.
+Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+
+The page will reload when you make changes.
+You may also see any lint errors in the console.
+
+### Setup
```
Comment (Meltrust) [inline README.md:105]: Your readme setup instructions are missing several things:

- [ ] Kindly, include in your instructions the required db commands that your users will need to run your backend app: ``rails db:create, rails db:migrate, rails db:seed``. 👍

- [ ] The seeds.rb is not working, it runs but won't create any new hotel records:

image

Kindly correct the seeds file, so your users can enjoy a prefilled database. 👍 

- [ ] Please, add a "tests" section with specific instructions about how to run your tests.
- [ ] Remove the frontend instructions (like npm start).  This is the backend project.

The project's documentation is important. Kindly be specific and ensure that _anyone_ can run your project just by following your instructions. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @gaks1,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  API is working well✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 3 above)
Comment (Koshao-sys) [reply README.md]: This is updated correctly.

File: README.md (hunk 2 above)
Comment (Koshao-sys) [reply README.md]: This is updated

**Review by Meltrust — APPROVED** [review]
#### Hi @gaks1,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Tests are now passing with updated instructions✔️ 
- Not used test folder has been removed✔️ 
- Instructions now pertain to the backend✔️
- seeds file has been fixed ✔️ 




Cheers and Happy coding!👏👏👏


------

**Pull request merged**